src/components/LeftSidebar/ConversationsList/ConversationsListVirtual.vue

```diff
@@ -3,153 +3,130 @@
-		},
-
-		/**
-		 * Scroll to conversation by index
-		 *
-		 * @public
-		 * @param {number} index - index of conversation to scroll to
-		 * @return {Promise<void>}
-		 */
-		async scrollToItem(index) {
-			const firstItemIndex = this.getFirstItemInViewportIndex()
-			const lastItemIndex = this.getLastItemInViewportIndex()
-
-			const viewportHeight = this.$refs.scroller.$el.clientHeight
-
-			/**
-			 * Scroll to a position with smooth scroll imitation
-			 *
-			 * @param {number} to - target position
-			 * @return {void}
-			 */
-			const doScroll = (to) => {
-				const ITEMS_TO_BORDER_AFTER_SCROLL = 1
-				const padding = ITEMS_TO_BORDER_AFTER_SCROLL * this.itemSize
-				const from = this.$refs.scroller.$el.scrollTop
-				const direction = from < to ? 1 : -1
-
-				// If we are far from the target - instantly scroll to a close position
-				if (Math.abs(from - to) > viewportHeight) {
-					this.$refs.scroller.scrollToPosition(to - direction * viewportHeight)
-				}
-
-				// Scroll to the target with smooth scroll
-				this.$refs.scroller.$el.scrollTo({
-					top: to + padding * direction,
-					behavior: 'smooth',
-				})
-			}
-
-			if (index < firstItemIndex) { // Item is above
-				await doScroll(index * this.itemSize)
-			} else if (index > lastItemIndex) { // Item is below
-				// Position of item + item's height and move to bottom
-				await doScroll((index + 1) * this.itemSize - viewportHeight)
-			}
-		},
-
-		/**
-		 * Scroll to conversation by token
-		 *
-		 * @param {string} token - token of conversation to scroll to
-		 * @return {void}
-		 */
-		scrollToConversation(token) {
-			const index = this.conversations.findIndex((conversation) => conversation.token === token)
-			if (index !== -1) {
-				this.scrollToItem(index)
-			}
-		},
-	},
+
+		// Scroll to the target with smooth scroll
+		containerProps.ref.value!.scrollTo({
+			top: to + padding * direction,
+			behavior: 'smooth',
+		})
+	}
+
+	if (index < firstItemIndex) { // Item is above
+		doScroll(index * itemHeight.value)
+	} else if (index > lastItemIndex) { // Item is below
+		// Position of item + item's height and move to bottom
+		doScroll((index + 1) * itemHeight.value - viewportHeight)
+	}
 }
+
+/**
+ * Scroll to conversation by token
+ *
+ * @param token - token of conversation to scroll to
+ */
+function scrollToConversation(token: string) {
+	const index = props.conversations.findIndex((conversation) => conversation.token === token)
+	if (index !== -1) {
+		scrollToItem(index)
+	}
+}
+
+defineExpose({
+	getFirstItemInViewportIndex,
+	getLastItemInViewportIndex,
+	scrollToItem,
+	scrollToConversation,
+})
 </script>
 
+<template>
+	<li v-bind="containerProps">
+		<LoadingPlaceholder v-if="loading" type="conversations" />
+		<ul
+			v-else
+			v-bind="wrapperProps">
+			<ConversationItem
+				v-for="item in list"
+				:key="item.data.id"
+				:item="item.data"
+				:compact="compact" />
```

Comment:
Just for the modern Vue style :P
```suggestion
				:compact />
```